clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -693,11 +708,33 @@
 
     public static final Schema[] UPDATE_METADATA_REQUEST = new Schema[] {UPDATE_METADATA_REQUEST_V0, UPDATE_METADATA_REQUEST_V1};
     public static final Schema[] UPDATE_METADATA_RESPONSE = new Schema[] {UPDATE_METADATA_RESPONSE_V0, UPDATE_METADATA_RESPONSE_V1};
+    public static final short[] DEPRECATED_UPDATE_METADATA_REQUEST = new short[] {};
+
+    /* Protocol version api */
+    public static final Schema PROTOCOL_VERSION_REQUEST_V0 = new Schema();
```

Comment:
Sure that is possible, all we will gain is saving few bytes over network for each connection. Clients will have to pro-actively know what APIs it is interested in.

I do not have a strong feeling in favor of against this. If multiple people find this helpful, I will make the change. Makes sense?